helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestServer.java

```diff
@@ -164,7 +165,10 @@ protected ResourceConfig getResourceConfig(HelixRestNamespace namespace, Servlet
     }
     cfg.property(ContextPropertyKeys.METADATA.name(), namespace);
 
-    cfg.register(new CORSFilter());
+    if (Boolean.getBoolean(CORS_ENABLED)) {
```

Comment:
Are you trying to get it from the system property? This always give you false.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html#getBoolean-java.lang.String-

It gets it from System Properties by the string key provided. So if the key-value pair is defined in System.Properties, this will evaluate to `true`